inc/saihostintf.h

```diff
@@ -212,101 +217,116 @@ typedef enum _sai_hostif_trap_type_t
+    SAI_HOSTIF_TRAP_TYPE_IP2ME = 0x00004000,
 
     /**
      * @brief SSH traffic (tcp dst port == 22) to local router IP address
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_SSH = 0x00002011,
+    SAI_HOSTIF_TRAP_TYPE_SSH = 0x00004001,
 
     /**
      * @brief SNMP traffic (udp dst port == 161) to local router IP address
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_SNMP = 0x00002012,
+    SAI_HOSTIF_TRAP_TYPE_SNMP = 0x00004002,
 
-    /** Router traps custom range start */
-    SAI_HOSTIF_TRAP_TYPE_ROUTER_CUSTOM_RANGE_BASE = 0x0003000,
+    /**
+    * @brief BGP traffic (tcp src port == 179 or tcp dst port == 179) to local
+    * router IP address (default packet action is drop)
+    */
+    SAI_HOSTIF_TRAP_TYPE_BGP = 0x00004003,
+
+    /**
+    * @brief BGPv6 traffic (tcp src port == 179 or tcp dst port == 179) to
+    * local router IP address (default packet action is drop)
+    */
+    SAI_HOSTIF_TRAP_TYPE_BGPV6 = 0x00004004,
+
+    /** Local IP traps custom range start */
+    SAI_HOSTIF_TRAP_TYPE_LOCAL_IP_CUSTOM_RANGE_BASE = 0x0005000,
 
     /* Pipeline exceptions */
 
     /**
      * @brief Packets size exceeds the router interface MTU size
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_L3_MTU_ERROR = 0x00004000,
+    SAI_HOSTIF_TRAP_TYPE_L3_MTU_ERROR = 0x00006000,
 
     /**
      * @brief Packets with TTL 0 or 1
-     * (default packet action is trap)
+     * (default packet action is drop)
      */
-    SAI_HOSTIF_TRAP_TYPE_TTL_ERROR = 0x00004001,
+    SAI_HOSTIF_TRAP_TYPE_TTL_ERROR = 0x00006001,
 
     /** Exception traps custom range start */
-    SAI_HOSTIF_TRAP_TYPE_CUSTOM_EXCEPTION_RANGE_BASE = 0x00005000,
+    SAI_HOSTIF_TRAP_TYPE_CUSTOM_EXCEPTION_RANGE_BASE = 0x00007000,
 
-} sai_hostif_trap_type_t;
+    /* User defined traps */
 
-/**
- * @brief Host interface trap channel
- */
-typedef enum _sai_hostif_trap_channel_t
-{
-    /** receive packets via file desriptor */
-    SAI_HOSTIF_TRAP_CHANNEL_FD,
+    /** router traps (default packet action is drop) */
+    SAI_HOSTIF_TRAP_TYPE_ROUTER_MIN = 0x00008000,
 
-    /** receive packets via callback */
-    SAI_HOSTIF_TRAP_CHANNEL_CB,
+    /** router traps max */
+    SAI_HOSTIF_TRAP_TYPE_ROUTER_MAX = SAI_HOSTIF_TRAP_TYPE_ROUTER_MIN + SAI_HOSTIF_TRAP_TYPE_TABLE_RANGE,
+
+    /** neighbor table traps (default packet action is drop) */
+    SAI_HOSTIF_TRAP_TYPE_NEIGH_MIN = SAI_HOSTIF_TRAP_TYPE_ROUTER_MAX + 1,
+
+    /** neighbor table traps max */
+    SAI_HOSTIF_TRAP_TYPE_NEIGH_MAX = SAI_HOSTIF_TRAP_TYPE_NEIGH_MIN + SAI_HOSTIF_TRAP_TYPE_TABLE_RANGE,
 
-    /** receive packets via OS net device */
-    SAI_HOSTIF_TRAP_CHANNEL_NETDEV,
+    /** ACL traps (default packet action is drop) */
+    SAI_HOSTIF_TRAP_TYPE_ACL_MIN = SAI_HOSTIF_TRAP_TYPE_NEIGH_MAX + 1,
```

Comment:
For each type (Router, neigh, ACL, FDB)
We allow SAI_HOSTIF_TRAP_TYPE_TABLE_RANGE
So the MAX trap ID for the type is MIN+SAI_HOSTIF_TRAP_TYPE_TABLE_RANGE
Then the next type is MAX+1